### ISSUE TITLE
Lockdown sensitive data throughout plugin for give_worker role

### ISSUE THREAD

**Issue by DevinWalker** [WordImpress/Give#1223]
## Issue Overview
As @Benunc pointed out, some sensitive data is displayed to users with the `give_worker` role. We need to ensure that this role does not have access to data it should not.

## Expected Behavior
If an admin sets a user's role as `give_worker` sensitive data should not be displayed to them.

## Current Behavior
In some cases, certain data is shown that shouldn't.

## Possible Solution
Perform necessary checks prior to output of sensitive data.

## Steps to Reproduce (for bugs)
1. Login to a Give dev site with donations and report data
2. Create a user with the `give_worker` role - tip: the "User Role Editor" plugin is extrememly helpful here: https://wordpress.org/plugins/user-role-editor/
3. View sensitive data

## Related PRs
List related PRs against other branches: https://github.com/WordImpress/Give/issues/662 https://github.com/WordImpress/Give/pull/1195

## Todos
- [ ] Audit sensitive data 
- [ ] Create list of data items to lock down
- [ ] Implement role checks
- [ ] Test
- [ ] Submit PR

**Comment by mehul0810** [comment]
@DevinWalker I've investigated some of the sensitive data displayed for `give_worker` user role.

**Here is the list of sensitive data items displayed to `give_worker` user role:**
1. Name (First Name & Last Name)
2. Email
3. Address

Also, noticed that `give_worker` user role is having an ability to to change donor or add new donor for existing donor. Let me know whether this also should be locked down?

### Reference
Screenshot 1:
<img width="1091" alt="sensitive_data_worker_role_1" src="https://user-images.githubusercontent.com/1852711/30483006-e2754fd8-9a42-11e7-8d97-f2906de5d2ce.png">

Screenshot 2:
<img width="1082" alt="sensitive_data_worker_role_2" src="https://user-images.githubusercontent.com/1852711/30482968-bc429d20-9a42-11e7-8816-37421f3fd112.png">

Let me know your insights on this.

**Comment by DevinWalker** [comment]
Thanks @mehul0810 - let's put this on hold until @mathetos delivers us his proposal on Give roles and their capabilities.

**Comment by mathetos** [comment]
**TL;DR** Give Workers should be able to see that info, but not change it. But we don't have a capability for that purpose yet.

Currently the Give Worker can indeed see all the Donations and edit them. They cannot however see the Donors, Reports, Settings, Tools. If we think of the Give Worker as an assistant to the Accountant and/or the Manager then what functions would they actually do with this data if they couldn't see the donor information? Honestly, they wouldn't really be able to help hardly anyone that way.

I could imagine the Accountant wanting them to look up a donation to confirm the form and amount. Or the donor themselves calling in to confirm their donation or get the email receipt re-sent. Those are tasks the Give Worker would be great for but it requires that they can see the donor information on the donation details screen as they do currently. 

The only change I would suggest here is that they only be able to VIEW the donation data, and not edit it at all. Editing donation data should be reserved for the Accountant and/or Manager roles. But currently we only have a capability called `read_private_give_payments` which currently doesn't provide read-only access to the Donations screen. That might just be a matter of updating the coding for the donations screen, or we might have to add a `read_give_payments` capability and apply that to the Worker, Account, and Manager.

**Comment by DevinWalker** [comment]
I'm closing this due to work being done on #2112
